### ISSUE TITLE
Assert failure: !\"Malformed m_pEEClass in MethodTable\"

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#25346]
**Job:**
https://mc.dot.net/#/user/coreclr-outerloop-gcstress0x3-gcstress0xc/ci~2Fdotnet~2Fcoreclr~2Frefs~2Fheads~2Fmaster/test~2Ffunctional~2Fcli~2F/20190623.1/workItem/JIT.Methodical/analysis/xunit/JIT_Methodical._VT_callconv__il_dbgjumps1__il_dbgjumps1_~2F_VT_callconv__il_dbgjumps1__il_dbgjumps1_cmd

**Failed tests:**
JIT_Methodical._VT_callconv__il_dbgjumps1__il_dbgjumps1_._VT_callconv__il_dbgjumps1__il_dbgjumps1_cmd

**Log:**
```
https://mc.dot.net/#/user/coreclr-outerloop-gcstress0x3-gcstress0xc/ci~2Fdotnet~2Fcoreclr~2Frefs~2Fheads~2Fmaster/test~2Ffunctional~2Fcli~2F/20190623.1/workItem/JIT.Methodical/analysis/xunit/JIT_Methodical._VT_callconv__il_dbgjumps1__il_dbgjumps1_~2F_VT_callconv__il_dbgjumps1__il_dbgjumps1_cmd
```

**Comment by AndyAyersMS** [comment]
This also might have been fixed already by #25333. Investigating.

**Comment by AndyAyersMS** [comment]
Nope, not fixed.

**Comment by AndyAyersMS** [comment]
Suspect this is caused by `emitter::emitEndFnEpilog()` underestimating the epilog size -- it is taking the minimum size (as the gc encoder can only handle one epilog size) -- and expecting there to just be one final instruction (which is why underestimating works, so long as the PC is within the final instruction... But now we can have two instructions, and that no longer works.

Still need to verify this is the issue, but seems likely. If so we should fix x86 to just use `jmp [addr]` like it did before, since x86 doesn't have limited displacement issues like x64 does.

This would also fix #25345, since we'd no longer emit `jmp [eax]`.

**Comment by xiangzhai** [comment]
:mips-interest
